content/docs/why-is-document-or-window-not-defined-in-client-component.mdx

```diff
@@ -0,0 +1,63 @@
+---
+updated: 2023-09-14
+title: 
+    "Why is document, window, localStorage or sessionStorage undefined in client components?"
+messageContent:
+    "Don't use document, window, localStorage or sessionStorage in client components outside of
+  useEffect. See the reason why and some code examples [here](<link>)."
+---
+
+> I'm having this function in client component
+>
+> ```ts
+> "use client";
+> export default function Page() {
+>   const [theme, setTheme] = useState(localStorage.getItem("theme"));
+> }
+> ```
+>
+> But I'm getting an error saying  
+> `ReferenceError: localStorage is not defined`
+>
+> Why is that? Isn't it supposed to work in client components?
+
+The name client component is quite misleading. It's not a component that runs in the browser. It's a
+component that runs on the **server** and the client. The code inside the component is executed
+first on the **server** and the client or in other words, it's pre rendered on the server and then
+hydrated on the client.
+
+### What's the fix for this?
+
+There are 2 fixes of this problem:
+
+- move logic using these APIs in `useEffect` hook.
+- import a component dynamically with `ssr:false`
+
+### Moving logic in useEffect
+
+If we move the logic inside `useEffect` hook, it will only run on the client. So, we can use these
+APIs without any problem.
+
+```ts
+"use client";
+export default function Page() {
+  const [theme, setTheme] = useState(undefined);
+
+  useEffect(() => {
+    setTheme(localStorage.getItem("theme"));
+  }, []);
+}
+```
+
+### Importing component dynamically
+
+If we import a component dynamically with `ssr:false`, it will only run on the client. So, we can
```

Comment:
```suggestion
If we [import a component dynamically with `ssr: false`](https://nextjs.org/docs/app/building-your-application/optimizing/lazy-loading#skipping-ssr), it will only run on the client, at the cost of the content not being available in the initial HTML (so crawlers may not be able to see it). If that is fine for you, you can proceed to
```